session/cache.go

```diff
@@ -0,0 +1,74 @@
+package session
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+)
+
+// Endpoint is a UDP endpoint in the discv5 protocol, consisting of a Node ID and UPD port number.
+type Endpoint struct {
+	NodeID uint64
+	Port   uint16
+}
+
+func (e Endpoint) String() string {
+	return fmt.Sprintf("%d:%d", e.NodeID, e.Port)
+}
+
+type cacheEntry struct {
+	k Endpoint
+	v interface{} // replace with session info struct
+}
+
+type Cache struct {
+	mu         sync.Mutex
+	lru        *list.List // Doubly-linked for storing items in LRU order
+	size       int
+	entriesMap map[Endpoint]*list.Element
+}
+
+const defaultMaxSize int = 100
+
+func NewCache() *Cache {
+	return &Cache{
+		lru:        list.New(),
+		size:       defaultMaxSize,
+		entriesMap: make(map[Endpoint]*list.Element),
+	}
+}
+
+func (c *Cache) StoreSession(endpoint Endpoint, session interface{}) {
```

Comment:
I think the `Session` suffix is a bit redundant. The call sites should be something like

```go
sessionCache.Store(endpoint, session)
```
and
```go
session, ok := sessionCache.Get(endpoint)
```

Reply:
Will do!